esrally/track/params.py

```diff
@@ -838,6 +838,19 @@ def params(self):
         return parsed_params
 
 
+class DownsampleParamSource(ParamSource):
+    def __init__(self, track, params, **kwargs):
+        super().__init__(track, params, **kwargs)
+        self._fixed_interval = params.get("fixed-interval", "1h")
+        self._source_index = params.get("source-index", "tsdb-index")
```

Comment:
Does a static default make sense here? We have a helper function for this purpose to extract a default target from whatever track we are running. Note: this would require the source parameter name be either `index` or `data-stream` (or support both, if it's supported by the API?)

```suggestion
        self._source_index = params.get_target(track, params)
```

Reply:
`source-index` and `target-index` is the same name we use in the docs about the downsampling API. I thought that keeping the naming consistent helps because they match and it is easier to understand what they refer to if you read the API docs.

Reply:
Got it. I'd still recommend this change, perhaps setting `params["index"] = params.get("source-index")` to use the function as-is, or replicating its functionality in-line to get a dynamic default

Reply:
The thing is...this API involves two indices...one that we read and one that we write. Having a parameter called `index`, in my opinion, does not make clear whether `index` is the source or the target.

Reply:
This would be transparent to the end user. just suggesting 
```suggestion
        # alias provided user param for get_target function
        params["index"] = params.get("source-index")
        self._source_index = params.get_target(track, params)
```

Reply:
or else, as I mentioned, just re-implement the logic for a default target from the function to here

Reply:
Moreover I see that `get_target` would support a parameter called `indices` which suggests this operation can be applied to multiple indices together. This is definitely not the case. A downsampling operation can only operate on a single source inde and on a single target index. Also data streams are not supported (unless at some point we decide some way to select the backing index in the data stream we would like to use as the source index).

Reply:
`indices` in this context is not a parameter but rather the track element, for example: https://github.com/elastic/rally-tracks/blob/master/tsdb/track.json#L21.

I think for some better information I'll reference the docs where this function is used elsewhere. From [Track Reference#search](https://esrally.readthedocs.io/en/stable/track.html#search):
```
index (optional): An [index pattern](https://www.elastic.co/guide/en/elasticsearch/reference/current/multi-index.html) that defines which indices or data streams should be targeted by this query. Only needed if the indices or data-streams section contains more than one index or data stream respectively. Otherwise, Rally will automatically derive the index or data stream to use. If you have defined multiple indices or data streams and want to query all of them, just specify "index": "_all".
```

In this operation's case it would be more
```
source-index (optional): The index containing data to aggregate which includes a ``@timestamp`` field. Note that this index should be marked read-only prior to the execution of this operation. Only needed if the indices section contains more than one index. Otherwise, Rally will automatically derive the index or data stream to use.
```